sgx-hasher/src/lib.rs

```diff
@@ -77,6 +81,68 @@ impl Hasher {
     }
 }
 
+pub fn to_array(bn: bn::BigNum) -> [u8; 384] {
+    let mut array = [0u8; 384];
+    array.copy_from_slice(&bn.to_vec());
+    array
+}
+
+pub fn sign(hash: &[u8; 32], key: &rsa::Rsa<Private>) -> Result<sig::Signature> {
+    let author = sig::Author::new(sig::Vendor::default(), 0u32, 0u32);
+
+    let contents = sig::Contents::new(
+        misc::MiscSelect::EXINFO,
+        attr::Attributes::new(attr::Flags::MODE_64_BIT, attr::Xfrm::all()),
+        *hash,
+        0,
+        0,
+    );
+
+    let rsa_key = PKey::from_rsa(key.to_owned())?;
+    let md = hash::MessageDigest::sha256();
+    let mut signer = sign::Signer::new(md, &rsa_key)?;
+    signer.update(author.as_ref())?;
+    signer.update(contents.as_ref())?;
+    let signature = signer.sign_to_vec()?;
+
+    let e = hex::encode(key.e().to_vec());
+    let e = u32::from_str_radix(&e[..], 16).unwrap();
```

Comment:
Don't use `hex` for this. We might want to get upstream `openssl` to `impl TryFrom<BigNum> for (u8, u16, u32, u64, u128, usize)`.